reactivesocket-cpp/src/Frame.cpp

```diff
@@ -114,34 +117,83 @@ std::ostream& operator<<(std::ostream& os, const FrameHeader& header) {
   std::bitset<16> flags(header.flags_);
   return os << header.type_ << "[" << flags << ", " << header.streamId_ << "]";
 }
+
+std::unique_ptr<folly::IOBuf> FrameMetadata::serialize() {
+  auto buf = FrameBufferAllocator::allocate(sizeof(uint32_t));
+  folly::io::Appender app(buf.get(), /* do not grow */ 0);
+  app.writeBE<uint32_t>(static_cast<uint32_t>(metadataPayload_->length()) + sizeof(uint32_t));
```

Comment:
nit: I doubt we will hit this, but seems prudent to assert that 

metadataPayload_->length() <= std::numeric_limits<int32_t>::max() - 1